packages/docusaurus/src/client/exports/Portal.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useRef, useState, useEffect, type ReactPortal} from 'react';
+import {createPortal} from 'react-dom';
+
+type PortalProps = {
+  children: React.ReactNode;
+  type: string;
+};
+
+export function Portal({
+  children,
+  type = 'docusaurus-portal',
```

Comment:
are other portal types planned? 

What is the purpose of calling `document.createElement("docusaurus-portal");` ?

Reply:
@slorber This is the container node's tag name. Portal elements can often be custom elements so as not to have any default user-agent styles.